### ISSUE TITLE
How to visualize the segmentation network performance output ?

### ISSUE THREAD

**Issue by arunumd** [yanx27/Pointnet_Pointnet2_pytorch#25]
Hi @yanx27 first of all thank you for this helpful work. I was first following the official tensorflow repository by the original authors. But it didn't work well for me. I later moved to your repository (PyTorch implementation) and I was able to do the training process without issues. I just have one question about the visualization. I was of the assumption that the **show3d_balls.py** file is supposed to help us visualize the output of segmentation like the way you have shown in the README.md file. But I find that the visualization code is not complete yet. Please correct me if I am wrong. What did you do to visualize the results of training, the way you have shown in the README.md file ? Did you use any external tools ?

**Comment by neverstoplearn** [comment]
i meet the same problem have you  solve it? thanks.

**Comment by arunumd** [comment]
I was not able to find out a way to visualize the results using this repository. I have found a more helpful repository where you just set the model path and other essential arguments to the visualizer and visualize it. The repository I am talking about is this one here => https://github.com/fxia22/pointnet.pytorch.git

**Comment by XGuider** [comment]
Hi, visual code was not found? Can you be more detailed?---->https://github.com/fxia22/pointnet.pytorch.git

**Comment by XGuider** [comment]
@arunumd

**Comment by BlurryLight** [comment]
@XGuider
https://github.com/fxia22/pointnet.pytorch/blob/master/utils/show_seg.py
it's not hard to follow that file and the code in this project to write a visualizer. What you need to do is to get the **points pred result**,namely `pred_choice` in code, with the `eval` model and draw it using the `showpoints` function.
https://github.com/yanx27/Pointnet_Pointnet2_pytorch/blob/31deedb10b85ec30178df57a6389b2f326f7c970/utils.py#L98

My code is here but it still needs to be modified to work well with your code.
https://github.com/BlurryLight/Pointnet_Pointnet2_pytorch/blob/master/visualizer/show_part_seg_pointnet.py

**Comment by XGuider** [comment]
yes，thnx you so much! i get .




------------------ Original ------------------
